static/app/components/organizations/pageFilters/container.tsx

```diff
@@ -125,33 +128,35 @@ function Container({skipLoadLastUsed, children, ...props}: Props) {
       return;
     }
 
-    const oldQuery = getStateFromQuery(lastQuery.current, {
+    const oldState = getStateFromQuery(lastQuery.current, {
       allowEmptyPeriod: true,
     });
-    const newQuery = getStateFromQuery(location.query, {
+    const newState = getStateFromQuery(location.query, {
       allowEmptyPeriod: true,
     });
 
-    const newEnvironments = newQuery.environment || [];
-    const newDateObject = getDateObjectFromQuery(newQuery);
-    const oldDateObject = getDateObjectFromQuery(oldQuery);
+    const newEnvironments = newState.environment || [];
+    const newDateState = getDatetimeFromState(newState);
+    const oldDateState = getDatetimeFromState(oldState);
 
-    /**
-     * Do not pass router to these actionCreators, as we do not want to update
-     * routes since these state changes are happening due to a change of routes
-     */
-    if (!isEqual(oldQuery.project, newQuery.project)) {
-      updateProjects(newQuery.project || [], null, {environments: newEnvironments});
-    } else if (!isEqual(oldQuery.environment, newQuery.environment)) {
-      /**
-       * When the project stays the same, it's still possible that the environment
-       * changed, so explictly update the enviornment
-       */
+    const noProjectChange = isEqual(oldState.project, newState.project);
+    const noEnvironmentChange = isEqual(oldState.environment, newState.environment);
+    const noDatetimeChange = isEqual(oldDateState, newDateState);
+
+    // Do not pass router to these actionCreators, as we do not want to update
+    // routes since these state changes are happening due to a change of routes
+    if (!noProjectChange) {
```

Comment:
If you wanted to avoid a double negation you could make the variables `sameProjects`, `sameEnvironment`, and `sameDatetime`.

Reply:
I think I'm OK with this, since it matches all the other variables above